github_stargazers/github_stargazers.py

```diff
@@ -7,53 +7,37 @@
 from github_stargazers.github import UsernameRepositoryError, TooManyRequestsHttpError, HTTPError
 
 
+class OutputPrintable(object):
+    @staticmethod
+    def print_stargazers(stargazers) -> None:
+        print("Stargazers:")
+        for stargazer in stargazers:
+            print(stargazer)
+
+    @staticmethod
+    def print_check_stargazer(is_stargazer) -> None:
+        return Halo().succeed("Stargazer") if is_stargazer else Halo().fail("Not a Stargazer")
+
+
 class _Command:  # pylint: disable=too-few-public-methods
 
     def __init__(self, username_and_repository: str, user: str) -> None:
         self.__username_and_repository: str = username_and_repository
         self.__user: typing.Optional[str] = user
 
-    def __get_github(self) -> typing.Optional[GitHub]:
-        try:
-            github = GitHub(self.__username_and_repository)
-        except UsernameRepositoryError as exception_message:
-            Halo().fail(exception_message)
-            return None
-        return github
-
-    @classmethod
-    def __print_stargazers(cls, github: GitHub) -> None:
-        assert github, "github cannot be None"
-        try:
-            stargazers: typing.List[str] = github.get_all_stargazers()
-        except (TooManyRequestsHttpError, HTTPError) as exception_message:
-            Halo().fail(exception_message)
-            return None
-        if not stargazers:
-            return None
-        print("Stargazers:")
-        for stargazer in stargazers:
-            print(stargazer)
 
-    def __print_check_stargazer(self, github: GitHub) -> None:
+    def process(self) -> None:
         try:
-            stargazer: bool = github.is_stargazer(self.__user)
-        except (TooManyRequestsHttpError, HTTPError) as exception_message:
+            github = GitHub(self.__username_and_repository)
```

Comment:
What do you think about treating `github = GitHub(self.__username_and_repository)` in a separate `try catch` block? The reason for this is that this statement can raise _only_ `UsernameRepositoryError`, while the other two can both raise `TooManyRequestsHttpError, HTTPError`.

Reply:
In this case maybe we should keep the `__get_github()` method because it does exactly what I described above, so we could keep that method from how it was before if you agree. Waiting for your opinion on this aspect.